src/test/java/com/amazon/opendistroforelasticsearch/security/test/helper/cluster/ClusterHelper.java

```diff
@@ -97,18 +97,24 @@ public String getClusterName() {
      */
 
     public final ClusterInfo startCluster(final NodeSettingsSupplier nodeSettingsSupplier, ClusterConfiguration clusterConfiguration) throws Exception {
-        return startCluster(nodeSettingsSupplier, clusterConfiguration, 10, null);
+        return startCluster(nodeSettingsSupplier, clusterConfiguration, 10, null, false);
     }
 
+    public final ClusterInfo restartCluster(final NodeSettingsSupplier nodeSettingsSupplier, ClusterConfiguration clusterConfiguration) throws Exception {
+        stopClusterWithoutDeletingData();
+        return startCluster(nodeSettingsSupplier, clusterConfiguration, 10, null, true);
+    }
 
-    public final synchronized ClusterInfo startCluster(final NodeSettingsSupplier nodeSettingsSupplier, ClusterConfiguration clusterConfiguration, int timeout, Integer nodes)
+    public final synchronized ClusterInfo startCluster(final NodeSettingsSupplier nodeSettingsSupplier, ClusterConfiguration clusterConfiguration, int timeout, Integer nodes, boolean isRestart)
```

Comment:
can it rely on `clusterInfo` to see if it is start (`null`) or restart (`!=null`)?

Reply:
ClusterInfo is not a member variable in ClusterHelper. So this class does not store any state that can be used to determine if its start or restart

Reply:
Can you introduce state into the `ClusterHelper` class that allows to track if it is started or stopped.

Reply:
Don't think that helps here. Restart is essentially stopping and starting again. How do we differentiate between the start transition of a first time start and a restart? We still need a function to restart and not just the state.

Reply:
Why does `ClusterHelper` need extra parameter `isRestart`? What if it is `true` on the first call to `startCluster()`? IMO, it is not needed and `ClusterHelper` should track cluster state. Another option is to use `ClusterInfo` to restart the cluster.

Reply:
Made the changes to add cluster state to ClusterHelper